CMakeLists.txt

```diff
@@ -1,19 +1,53 @@
-cmake_minimum_required(VERSION 2.8)
-project(mustache)
-if (UNIX)
-	add_definitions(
-		-Wall
-		-Wextra
-		-Werror
-		-std=c++11
-	)
-elseif (MSVC)
-	add_definitions(
-		/W3
-		/WX
-	)
+cmake_minimum_required(VERSION 3.7.2)
+
+project(mustache VERSION 4.0.0 LANGUAGES CXX)
+
+option(MUSTACHE_BUILD_TESTS
+       "build the tests with catch"
+       ON)
+
+add_library(${PROJECT_NAME}
+            INTERFACE)
+target_include_directories(${PROJECT_NAME}
+                           INTERFACE $<BUILD_INTERFACE:${${PROJECT_NAME}_SOURCE_DIR}/>
+                                     $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>)
+
+if(MUSTACHE_BUILD_TESTS)
+  if(UNIX)
+    add_definitions(-Wall
+                    -Wextra
+                    -Werror
+                    -std=c++11)
+  elseif(MSVC)
+    add_definitions(/W3
+                    /WX)
+  endif()
+  add_executable(${PROJECT_NAME}-test
+                 ${PROJECT_NAME}.hpp # to show in IDE
+                 tests.cpp)
 endif()
-add_executable(mustache
-	mustache.hpp # to show in IDE
-	tests.cpp
-)
+
+include(GNUInstallDirs)
+install(TARGETS ${PROJECT_NAME}
+        EXPORT ${PROJECT_NAME}_Targets
+        ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
+        LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+        RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR})
+install(FILES ${PROJECT_SOURCE_DIR}/${PROJECT_NAME}.hpp DESTINATION include/${PROJECT_NAME})
+install(EXPORT ${PROJECT_NAME}_Targets
+        FILE ${PROJECT_NAME}Targets.cmake
+        NAMESPACE ${PROJECT_NAME}::
```

Comment:
We should use `kainjow` as the namespace, since the C++ code already does this. I think people are already doing this in other uses of this library, so might as well make it more unique. So then include dir would be `kainjow/mustache.hpp` and not conflict with any other generic "mustache" library.